<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>optwndw</class>
 <widget class="QWidget" name="optwndw">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>338</width>
    <height>297</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Options ▲Δ</string>
  </property>
  <widget class="QPushButton" name="closeButton">
   <property name="geometry">
    <rect>
     <x>245</x>
     <y>265</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>close</string>
   </property>
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/48x48/png/48x48/Run.png</normaloff>:/48x48/png/48x48/Run.png</iconset>
   </property>
  </widget>
  <widget class="QPushButton" name="saveButton">
   <property name="geometry">
    <rect>
     <x>155</x>
     <y>265</y>
     <width>75</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>save</string>
   </property>
   <property name="icon">
    <iconset resource="../resources/resources.qrc">
     <normaloff>:/48x48/png/48x48/Save.png</normaloff>:/48x48/png/48x48/Save.png</iconset>
   </property>
  </widget>
  <widget class="QTabWidget" name="tabWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>326</width>
     <height>251</height>
    </rect>
   </property>
   <property name="tabPosition">
    <enum>QTabWidget::North</enum>
   </property>
   <property name="tabShape">
    <enum>QTabWidget::Triangular</enum>
   </property>
   <property name="currentIndex">
    <number>1</number>
   </property>
   <widget class="QWidget" name="tab">
    <attribute name="title">
     <string>Camera</string>
    </attribute>
    <widget class="QFrame" name="frame">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>321</width>
       <height>151</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="midLineWidth">
      <number>0</number>
     </property>
     <widget class="QSlider" name="cambright_sdr">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>75</y>
        <width>271</width>
        <height>19</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>webcam brightness</string>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksAbove</enum>
      </property>
      <property name="tickInterval">
       <number>25</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>55</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Webcam Brightness</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>15</y>
        <width>126</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Minimum pixel brightness</string>
      </property>
     </widget>
     <widget class="QSlider" name="pixbright_sdr">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>35</y>
        <width>271</width>
        <height>19</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>how bright a pixel must be at minimum to be detected</string>
      </property>
      <property name="minimum">
       <number>10</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksAbove</enum>
      </property>
      <property name="tickInterval">
       <number>25</number>
      </property>
     </widget>
     <widget class="QLabel" name="pixbright_lbl">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>15</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="cambright_lbl">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>55</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QSlider" name="exposure_sdr">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>115</y>
        <width>271</width>
        <height>19</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>5000</number>
      </property>
      <property name="pageStep">
       <number>1</number>
      </property>
      <property name="sliderPosition">
       <number>0</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="invertedAppearance">
       <bool>false</bool>
      </property>
      <property name="invertedControls">
       <bool>false</bool>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksAbove</enum>
      </property>
      <property name="tickInterval">
       <number>25</number>
      </property>
     </widget>
     <widget class="QLabel" name="exposure_lbl">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>95</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>25</x>
        <y>95</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Webcam Exposure</string>
      </property>
     </widget>
    </widget>
    <widget class="QComboBox" name="resoBox">
     <property name="geometry">
      <rect>
       <x>205</x>
       <y>195</y>
       <width>101</width>
       <height>22</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>-1</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>205</x>
       <y>170</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Resolution</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_2">
    <attribute name="title">
     <string>Scan</string>
    </attribute>
    <widget class="QFrame" name="frame_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>316</width>
       <height>231</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <property name="midLineWidth">
      <number>0</number>
     </property>
     <widget class="QLabel" name="scanrev_lbl">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>10</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>scans/rev</string>
      </property>
     </widget>
     <widget class="QSlider" name="scansrev_sdr">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>30</y>
        <width>281</width>
        <height>19</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>how many scans per revolution should we take?</string>
      </property>
      <property name="minimum">
       <number>100</number>
      </property>
      <property name="maximum">
       <number>400</number>
      </property>
      <property name="singleStep">
       <number>100</number>
      </property>
      <property name="pageStep">
       <number>100</number>
      </property>
      <property name="sliderPosition">
       <number>100</number>
      </property>
      <property name="tracking">
       <bool>true</bool>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="invertedAppearance">
       <bool>false</bool>
      </property>
      <property name="invertedControls">
       <bool>false</bool>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksAbove</enum>
      </property>
      <property name="tickInterval">
       <number>100</number>
      </property>
     </widget>
     <widget class="QLabel" name="steps_lbl">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>10</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QSlider" name="stepdelay_sdr">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>276</width>
        <height>19</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>this is the delay between each step; make it as low as possible to increase scan speed, 
but only as high as there are no scan errors because the cam needs time to settle</string>
      </property>
      <property name="minimum">
       <number>100</number>
      </property>
      <property name="maximum">
       <number>1200</number>
      </property>
      <property name="singleStep">
       <number>10</number>
      </property>
      <property name="pageStep">
       <number>50</number>
      </property>
      <property name="value">
       <number>200</number>
      </property>
      <property name="sliderPosition">
       <number>200</number>
      </property>
      <property name="tracking">
       <bool>true</bool>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="invertedAppearance">
       <bool>false</bool>
      </property>
      <property name="invertedControls">
       <bool>false</bool>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksAbove</enum>
      </property>
      <property name="tickInterval">
       <number>100</number>
      </property>
     </widget>
     <widget class="QLabel" name="stepdelay_lbl">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>50</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>stepdelay                    ms</string>
      </property>
     </widget>
     <widget class="QLabel" name="delay_lbl">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>50</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QFrame" name="frame_3">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>145</y>
        <width>191</width>
        <height>81</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <property name="midLineWidth">
       <number>0</number>
      </property>
      <widget class="QSpinBox" name="dspinBox">
       <property name="geometry">
        <rect>
         <x>76</x>
         <y>50</y>
         <width>51</width>
         <height>22</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>lower border of scan area</string>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="singleStep">
        <number>2</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="rspinBox">
       <property name="geometry">
        <rect>
         <x>135</x>
         <y>30</y>
         <width>51</width>
         <height>22</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>right border of scan area</string>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="singleStep">
        <number>2</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="lspinBox">
       <property name="geometry">
        <rect>
         <x>15</x>
         <y>30</y>
         <width>51</width>
         <height>22</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>left border of scan area</string>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="singleStep">
        <number>2</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="uspinBox">
       <property name="geometry">
        <rect>
         <x>76</x>
         <y>10</y>
         <width>51</width>
         <height>22</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>upper border of scan area</string>
       </property>
       <property name="frame">
        <bool>true</bool>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="singleStep">
        <number>2</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>65</x>
         <y>30</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Borders</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <zorder>label_5</zorder>
      <zorder>dspinBox</zorder>
      <zorder>rspinBox</zorder>
      <zorder>lspinBox</zorder>
      <zorder>uspinBox</zorder>
     </widget>
     <widget class="QDoubleSpinBox" name="r_angleBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>62</width>
        <height>22</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>this is the angle between the webcam and the right laser. (positive value)</string>
      </property>
      <property name="decimals">
       <number>2</number>
      </property>
      <property name="maximum">
       <double>90.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.500000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>95</y>
        <width>96</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>[°]Cam&gt;right laser</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="l_angleBox">
      <property name="geometry">
       <rect>
        <x>115</x>
        <y>120</y>
        <width>62</width>
        <height>22</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>this is the angle between the webcam and the left laser. (positive value)</string>
      </property>
      <property name="decimals">
       <number>2</number>
      </property>
      <property name="maximum">
       <double>90.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.500000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>115</x>
        <y>95</y>
        <width>96</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>[°]Cam&gt;left laser</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="cam_angleBox">
      <property name="geometry">
       <rect>
        <x>215</x>
        <y>120</y>
        <width>62</width>
        <height>22</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>this is the angle between the webcam and the turntable. (positive value)</string>
      </property>
      <property name="maximum">
       <double>90.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.500000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>215</x>
        <y>95</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>[°]Cam&gt;turntable</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_3">
    <attribute name="title">
     <string>Serial</string>
    </attribute>
    <widget class="QFrame" name="frame_4">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>316</width>
       <height>231</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QComboBox" name="comBox">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>25</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="toolTip">
       <string>comports are scanned when you open the program; 
if you changed the ports etc. you need a restart of the program to refresh the list.</string>
      </property>
     </widget>
     <widget class="QComboBox" name="baudBox">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>55</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>25</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="text">
       <string>Com Port</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::NoTextInteraction</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>55</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="text">
       <string>Baud</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::NoTextInteraction</set>
      </property>
     </widget>
     <widget class="QFrame" name="frame_7">
      <property name="geometry">
       <rect>
        <x>155</x>
        <y>20</y>
        <width>156</width>
        <height>61</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QPushButton" name="connectBtn">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>30</y>
         <width>71</width>
         <height>23</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>connect</string>
       </property>
       <property name="text">
        <string>Connect</string>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/myres/png/48x48/OK.png</normaloff>:/myres/png/48x48/OK.png</iconset>
       </property>
      </widget>
      <widget class="QCheckBox" name="connectBox">
       <property name="geometry">
        <rect>
         <x>15</x>
         <y>5</y>
         <width>91</width>
         <height>17</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>connect the comport at startup?</string>
       </property>
       <property name="text">
        <string>Auto Connect</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_4">
    <attribute name="title">
     <string>System</string>
    </attribute>
    <widget class="QFrame" name="frame_5">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>316</width>
       <height>231</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QPushButton" name="selectfileButton">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>31</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="icon">
       <iconset resource="../resources/resources.qrc">
        <normaloff>:/48x48/png/48x48/View.png</normaloff>:/48x48/png/48x48/View.png</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>24</width>
        <height>24</height>
       </size>
      </property>
     </widget>
     <widget class="QTextEdit" name="folderlabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>301</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>7</pointsize>
       </font>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:7pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;default&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>25</y>
        <width>211</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Please select your working directory</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources>
  <include location="../resources/resources.qrc"/>
 </resources>
 <connections/>
</ui>
